server.go

```diff
@@ -96,7 +96,7 @@ func (s *server) Allow(namespace, name string, tokensRequested int64, maxWaitMil
 	}
 
 	maxWaitTime := time.Millisecond
-	if maxWaitMillisOverride > -1 && maxWaitMillisOverride < b.Config().WaitTimeoutMillis {
+	if maxWaitTimeOverride {
 		// Use the max wait time override from the request.
 		maxWaitTime *= time.Duration(maxWaitMillisOverride)
```

Comment:
We still should check that `maxWaitMillisOverride` is less than `b.Config().WaitTimeoutMillis`